### ISSUE TITLE
Errors installing packages on aarch64?

### ISSUE THREAD

**Issue by jdomke** [spack/spack#12767]
current main branch (develop) cannot build any packages
(git hash 02931a82fd506a095ab93d6ad1d0dc416caf141d)

### Steps to reproduce the issue

```console
git clone https://github.com/spack/spack.git
. spack/share/spack/setup-env.sh
spack install hdf5
```

### Error Message

```console
==> Warning: CROSS has not been set, however cannot detect scheduler.
==> Error: Command exited with status 1:
'/usr/bin/clang' '-v' '/tmp/spack-implicit-link-infobHKHO2/main.c' '-o' '/tmp/spack-implicit-link-infobHKHO2/output'
clang version 3.4.2 (tags/RELEASE_34/dot2-final)
Target: aarch64-redhat-linux-gnu
Thread model: posix
Selected GCC installation: 
 "/usr/bin/clang" -cc1 -triple aarch64-redhat-linux-gnu -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -main-file-name main.c -mrelocation-model static -mdisable-fp-elim -fmath-errno -masm-verbose -mconstructor-aliases -fuse-init-array -target-cpu generic -target-linker-version 2.27 -v -resource-dir /usr/bin/../lib/clang/3.4.2 -internal-isystem /usr/local/include -internal-isystem /usr/bin/../lib/clang/3.4.2/include -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdebug-compilation-dir /home/jens -ferror-limit 19 -fmessage-length 0 -mstackrealign -fno-signed-char -fobjc-runtime=gcc -fdiagnostics-show-option -vectorize-slp -o /tmp/main-07f21b.o -x c /tmp/spack-implicit-link-infobHKHO2/main.c
clang -cc1 version 3.4.2 based upon LLVM 3.4.2 default target aarch64-redhat-linux-gnu
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /usr/bin/../lib/clang/3.4.2/include
 /usr/include
End of search list.
 "/usr/bin/ld" --hash-style=gnu --no-add-needed --build-id --eh-frame-hdr -m aarch64linux -dynamic-linker /lib/ld-linux-aarch64.so.1 -o /tmp/spack-implicit-link-infobHKHO2/output /lib/../lib64/crt1.o /lib/../lib64/crti.o crtbegin.o -L/lib/../lib64 -L/usr/lib/../lib64 -L/lib -L/usr/lib /tmp/main-07f21b.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed crtend.o /lib/../lib64/crtn.o
/usr/bin/ld: cannot find crtbegin.o: No such file or directory
/usr/bin/ld: cannot find -lgcc
/usr/bin/ld: cannot find -lgcc_s
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Possible mitigation

```
git checkout v0.12.1
```

### Information on your system

System: ThunderX2 (aarch64)
Linux: CentOS Linux release 7.6.1810
Kernel: 4.14.0-115.10.1.el7a.aarch64

**Comment by alalazo** [comment]
Can you bisect starting from a commit that is known to be good? I am wondering if this is related in some way to your configuration, but have no access so far to an ARM architecture.

**Comment by jdomke** [comment]
last good commit i know if is 4f9131fdc2 (between this and now i have not tried other versions)

**Comment by alalazo** [comment]
Can you use [git-bisect](https://git-scm.com/docs/git-bisect) to check where this starts? Just try to install hdf5 at every commit, and mark them good or bad depending on whether they suceed or not.

**Comment by jdomke** [comment]
ok, finished bisecting, last good commit is a7e9f477fcfdc5caac4585d7cbf0df85c1829b18

**Comment by alalazo** [comment]
Thanks, so probably we need to investigate #7153 i.e. the commit after that.

**Comment by alalazo** [comment]
It seems #7153 changes `compilers.yaml`. Did you try already to move your current `compilers.yaml` to some backup area and try to set up again your compilers? (Apologies if I can't be more precise than that upfront, but here I'm flying blind as I don't have any `aarch64` machine to log into).

**Comment by jdomke** [comment]
basically i cannot even create a new `compilers.yaml`
`spack compiler find` will already fail

**Comment by adamjstewart** [comment]
This should be fixed by #12756. Can you confirm?

**Comment by jdomke** [comment]
Yes, can confirm that `spack compiler find` and `spack install ...` worked again. Thanks.
